src/Input.cpp

```diff
@@ -250,58 +250,60 @@ static void ReadCmakelist(const Configuration& config, std::unordered_map<std::s
             comp.recreate = true;
             continue;
         }
-        if (line.size() > 0 && line[0] == '#') {
+        if ((line.size() > 0 && line[0] == '#') || line.empty()) {
             continue;
         }
-        int newParenLevel = parenLevel + std::count(line.begin(), line.end(), '(')
+        int newParentLevel = parentLevel + std::count(line.begin(), line.end(), '(')
                                 - std::count(line.begin(), line.end(), ')');
         if (strstr(line.c_str(), "project(") == line.c_str()) {
             size_t end = line.find(')');
             if (end != line.npos) {
                 comp.name = line.substr(8, end - 8);
             }
-        }
-        else if (TryGetComponentType(comp, config.addLibraryAliases, line)) {
+        } else if (TryGetComponentType(comp, config.addLibraryAliases, line)) {
             inTargetDefinition = true;
         } else if (TryGetComponentType(comp, config.addExecutableAliases, line)) {
             inTargetDefinition = true;
-        } else if (inTargetDefinition) {
-            const size_t endOfLine = (newParenLevel == 0) ? line.find_last_of(')')
-                                                          : line.length();
-            if (endOfLine > 0) {
-                const std::string targetLine(line.substr(0, endOfLine));
-                if (!strstr(targetLine.c_str(), "${IMPLEMENTATION_SOURCES}") &&
-                    !strstr(targetLine.c_str(), "${IMPLEMENTATION_HEADERS}") &&
-                    !IsCode(filesystem::path(targetLine).extension().generic_string())) {
-                    comp.additionalTargetParameters.append(targetLine + '\n');
-                }
-            }
-            if (newParenLevel == 0) {
-                inTargetDefinition = false;
-            }
         } else if (config.reuseCustomSections) {
             if (IsAutomaticlyGeneratedSection(line))
             {
                 inAutoSection = true;
             } else {
-                if (!inAutoSection && !line.empty()) {
+                if (!inAutoSection && !line.empty() && !inTargetDefinition) {
                     comp.additionalCmakeDeclarations.append(line + '\n');
                 }
             }
-            if (inAutoSection && newParenLevel == 0) {
+            if (inAutoSection && newParentLevel == 0) {
                 inAutoSection = false;
             }
         }
-        parenLevel = newParenLevel;
+        if (inTargetDefinition) {
+            const size_t endOfLine = (newParentLevel == 0) ? line.find_last_of(')')
+                                                           : line.length();
+            if (endOfLine > 0) {
+                const std::string targetLine(line.substr(0, endOfLine));
+                static const std::unordered_set<std::string> cmakeTargetParameters = { "ALIAS", "EXCLUDE_FROM_ALL", "GLOBAL", "IMPORTED", "INTERFACE", "MACOSX_BUNDLE", "MODULE", "OBJECT", "STATIC", "SHARED", "UNKNOWN", "WIN32" };
```

Comment:
Smells like it should be configurable, or at least something you can add on to. Also, some of these change the meaning of other arguments, for example you don't list files after ALIAS. Is this actually tested to work well?

Reply:
Not sure whether we want to have this set of options configurable it is highly depending on CMake itself. With regards of testing the inputs I don't think the cpp-dependencies should handle this at all, that is up to the CMake tool itself otherwise we will start to replicate all the logic here. If an user has already written an CMakeLists.txt we should expect that it is at least accepted by the CMake parser.

Reply:
Actually when checking again we should make set of options which are actually allowed in combination and we should restrict ourselves to specific CMake versions as some options are new for example. So we should put a bit more thinking in how far we want to go here.

For example even the test scenario of INTERFACE EXCLUDE_FROM_ALL is not a valid definition.

Reply:
Actually when checking again we should make set of options which are actually allowed in combination and we should restrict ourselves to specific CMake versions as some options are new for example. So we should put a bit more thinking in how far we want to go here.

For example even the test scenario of INTERFACE EXCLUDE_FROM_ALL is not a valid definition.